plasma/plasma-mvp/specifications/morevp.md

```diff
@@ -0,0 +1,655 @@
+The contract can then validate that these transactions are the correct ones, that they were included in the chain, that the signatures are correct, and that the exiting transaction is correctly formed.
+This proves the exitability of the transaction.
+2. The user may present the transaction along signatures the user claims to be valid.
+The contract can validate that the exiting transaction is correctly formed.
+Another user can challenge one of these signatures by presenting some transaction that created an input such that the true input owner did not sign the signature.
+In this case, the exit would be blocked entirely and the challenging user would receive `exit bond`.
+
+Option (1) checks that a transaction is exitable when the exit is started.
+This has lower communication cost and complexity but higher up-front gas cost.
+This option also ensures that only a single exit on any given transaction can exist at any point in time.
+Option (2) allows a user to assert that a transaction is exitable, but leaves the proof to a challenge-response game.
+This is cheaper up-front but adds complexity.
+This option must permit multiple exits on the same transaction, as some exits may provide invalid signatures.
+
+<!-- TODO: Groom this section once we decide on (1) or (2) -->
+
+These are not the only possible mechanisms that prove a transaction is exitable.
+There may be further ways to optimize these two options.
+
+We still need to provide a deterministic ordering of exits by some priority.
+MoreVP exits are given a priority based on the position in the Plasma chain of the most recently included (youngest) input to that transaction.
+Unlike the MVP protocol, we give each input and output to a transaction the same priority.
+This should be implemented by inserting a single “exit” object into a priority queue of exits and tracking a list of inputs or outputs to be paid out once the exit is processed.
+
+
+#### Proving Canonicity
+
+Whether unspent inputs or unspent outputs are paid out in an exit depends on the canonicity of the referenced transaction, independent of any piggybacking by other users.
+Unfortunately it’s too expensive to directly prove that a transaction is or is not canonical.
+Instead, we assume that the referenced transaction is canonical by default and allow a series of challenges and responses to determine the true canonicity of the transaction.
+
+The process of determining canonicity involves a challenge-response game.
+In the first period of the exit, any user may reveal a competing transaction that potentially makes the exiting transaction non-canonical.
+This competing transaction must be exitable and must share an input with the exiting transaction, but does not have to be included in the Plasma chain.
```

Comment:
add that the competing txn doesnt have to be included in the chain so to ensure that if users conduct other txns with the same input during an exit's challenge period, their youngest txn will take priority..

Reply:
I think the competing transaction doesn't have to be included in the chain because the exiting transaction might not be either.  In that case, neither is canonical.

> If the transaction was not included in the chain, an input to that transaction would be considered previously spent if another transaction also spending the input is *known to exist*.